service/sentry/sentry.go

```diff
@@ -0,0 +1,82 @@
+package sentry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/CanonicalLtd/serial-vault/service/response"
+	"github.com/getsentry/sentry-go"
+)
+
+const placeholder = "[Filtered]"
+
+var headerFilter = []string{"Api-Key", "Authorization", "User", "Cookie"}
+
+// Init initializes sentry client
+func Init(dsn, serviceName, serviceVersion string) error {
+	sentrySyncTransport := sentry.NewHTTPSyncTransport()
+	sentrySyncTransport.Timeout = time.Second * 3
+
+	opt := sentry.ClientOptions{
+		Dsn:              dsn,
+		Release:          serviceVersion,
+		Transport:        sentrySyncTransport,
+		AttachStacktrace: true,
+		BeforeSend:       getFilteredEvent,
+	}
+	if err := sentry.Init(opt); err != nil {
+		return err
+	}
+
+	sentry.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetTag("app_name", serviceName)
+	})
+	return nil
+}
+
+// Report creates sentry report from response.ErrorResponse
+func Report(ctx context.Context, resp response.ErrorResponse) {
+	if !shouldReport(resp) {
+		return
+	}
+
+	if hub := sentry.GetHubFromContext(ctx); hub != nil {
+		go hub.WithScope(func(scope *sentry.Scope) {
+			// create a sentry report
+			scope.SetTag("http_status_code", fmt.Sprint(resp.StatusCode))
+			scope.SetTag("error_code", resp.Code)
+			if resp.SubCode != "" {
+				scope.SetTag("error_subcode", resp.SubCode)
+			}
+			scope.SetLevel(sentry.LevelError)
+			hub.CaptureMessage(resp.Message)
+		})
+	}
+}
+
+func getFilteredEvent(event *sentry.Event, hint *sentry.EventHint) *sentry.Event {
+	for _, headerName := range headerFilter {
+		if _, ok := event.Request.Headers[headerName]; ok {
+			event.Request.Headers[headerName] = placeholder
+		}
+	}
+
+	if event.Request.QueryString != "" {
+		event.Request.QueryString = placeholder
+	}
+
+	if event.Request.Data != "" {
+		event.Request.Data = placeholder
+	}
+
+	if event.Request.Cookies != "" {
+		event.Request.Cookies = placeholder
```

Comment:
It might be useful to retain the names of the cookies and filter out the values?